### ISSUE TITLE
busser-serverspec cannot run against chef 11

### ISSUE THREAD

**Issue by jamesacle** [test-kitchen/busser-serverspec#45]
I have an older installation of Chef 11 I support on Oracle Enterprise Linux 6.4. My test kitchen tests recently stopped working because of the following issue:

```
       Installing Busser plugins: busser-serverspec
       /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/installer.rb:388:in `ensure_required_ruby_version_met': rake requires Ruby version >= 2.0.0. (Gem::InstallError)
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/installer.rb:156:in `install'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:297:in `block in install'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:270:in `each'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:270:in `each_with_index'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:270:in `install'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/rubygems.rb:44:in `install_gem'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:66:in `install_plugin_gem'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:53:in `install'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:43:in `block (2 levels) in install_all'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:43:in `each'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:43:in `block in install_all'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/rubygems.rb:71:in `silence_gem_ui'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:42:in `install_all'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/command.rb:27:in `run'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:126:in `invoke_command'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `block in invoke_all'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `each'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `map'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `invoke_all'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/group.rb:232:in `dispatch'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:115:in `invoke'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:40:in `block in register'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/command.rb:27:in `run'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:126:in `invoke_command'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:359:in `dispatch'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:115:in `invoke'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:235:in `block in subcommand'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/command.rb:27:in `run'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:126:in `invoke_command'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:359:in `dispatch'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/base.rb:440:in `start'
       	from /tmp/verifier/gems/gems/busser-0.7.1/bin/busser:8:in `<top (required)>'
       	from /tmp/verifier/bin/busser:23:in `load'
       	from /tmp/verifier/bin/busser:23:in `<main>'
```

Is it possible to pin rake to an older version for ruby 1.9.1?

**Comment by cheeseplus** [comment]
You can specify version in the Gemfile (as per the README) but that's about all that can be done there. Given Chef 11 and Ruby 1.9.1 being EOL, it's best to pin to older versions if you need to support these.

**Comment by jamesacle** [comment]
@cheeseplus  I didn't realize there was Gemfile support! Thank you, that seems like a reasonable solution!

**Comment by cheeseplus** [comment]
It was never advertised terribly well but you can basically pin the world there if needed, it can get hairy but definitely doable

**Comment by jamesacle** [comment]
@cheeseplus hey Seth - sorry to bother you on this old issue again, but I cannot for the life of me get the Gemfile support working. i'm wondering if my kitchen version is too old or something.

Anyway, here's what I have. I place a Gemfile in cookbooks/ci-streaming-mirror-maker/test/integration/mirror-maker-scripts/serverspec/Gemfile. The contents are:

`gem "busser", "0.7.0"`

I chose this because the busser Gem is resolved pretty early in the chain, and resolves to 0.7.1 by default.

Once my recipes have run, this is what I see in the  logs:

```
-----> Starting Kitchen (v1.10.0)
-----> Cleaning up any prior instances of <mirror-maker-scripts-oracle-6-4>
-----> Destroying <mirror-maker-scripts-oracle-6-4>...
       ==> default: Forcing shutdown of VM...
       ==> default: Destroying VM and associated drives...
       Vagrant instance <mirror-maker-scripts-oracle-6-4> destroyed.
       Finished destroying <mirror-maker-scripts-oracle-6-4> (0m3.22s).
-----> Testing <mirror-maker-scripts-oracle-6-4>
-----> Creating <mirror-maker-scripts-oracle-6-4>...
       Bringing machine 'default' up with 'virtualbox' provider...
       ==> default: Importing base box 'oracle_6_4'...
==> default: Matching MAC address for NAT networking...
       ==> default: Setting the name of the VM: kitchen-ci-streaming-mirror-maker-mirror-maker-scripts-oracle-6-4_default_1520546902704_66723
       ==> default: Fixed port collision for 22 => 2222. Now on port 2200.
       ==> default: Clearing any previously set network interfaces...
       ==> default: Preparing network interfaces based on configuration...
           default: Adapter 1: nat
       ==> default: Forwarding ports...
           default: 22 (guest) => 2200 (host) (adapter 1)
       ==> default: Booting VM...
       ==> default: Waiting for machine to boot. This may take a few minutes...
           default: SSH address: 127.0.0.1:2200
           default: SSH username: vagrant
           default: SSH auth method: private key
       ==> default: Machine booted and ready!
       ==> default: Checking for guest additions in VM...
       ==> default: Setting hostname...
       ==> default: Machine not provisioned because `--no-provision` is specified.
       [SSH] Established
       Vagrant instance <mirror-maker-scripts-oracle-6-4> created.
       Finished creating <mirror-maker-scripts-oracle-6-4> (0m39.55s).
-----> Converging <mirror-maker-scripts-oracle-6-4>...
       Preparing files for transfer
       Preparing dna.json
       Resolving cookbook dependencies with Berkshelf 3.2.0...
       Removing non-cookbook files before transfer
       Preparing data_bags
       Preparing environments
       Preparing validation.pem
       Preparing client.rb
-----> Chef Omnibus installation detected (install only if missing)
       Transferring files to <mirror-maker-scripts-oracle-6-4>
       [2018-03-08T22:08:51+00:00] WARN:
       * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
       SSL validation of HTTPS requests is disabled. HTTPS connections are still
       encrypted, but chef is not able to detect forged replies or man in the middle
       attacks.

       To fix this issue add an entry like this to your configuration file:

       ```
         # Verify all HTTPS connections (recommended)
         ssl_verify_mode :verify_peer

         # OR, Verify only connections to chef-server
         verify_api_cert true
       ```

       To check your SSL configuration, or troubleshoot errors, you can use the
       `knife ssl check` command like so:

       ```
         knife ssl check -c /tmp/kitchen/client.rb
       ```

       * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *

       Starting Chef Client, version 11.12.4
       Creating a new client identity for mirror-maker-scripts-oracle-6-4 using the validator key.
       resolving cookbooks for run list: ["ci-streaming-mirror-maker::deploy-kafka-frigg-mirrormaker-scripts"]
       Synchronizing Cookbooks:
         - ci-streaming-mirror-maker
         - ci-base
         - ci-deploy
         - java
         - ci-pipeline-streaming
         - ci-logdir
         - ci-log-rotation
         - sudo
         - yum
         - ulimit
         - line
         - ci-metrics-library
         - tar
       Compiling Cookbooks...
       Converging 4 resources

<recipes content removed, but they ran correctly>

       Running handlers:
       Running handlers complete
       Chef Client finished, 4/4 resources updated in 9.403789827 seconds
       Finished converging <mirror-maker-scripts-oracle-6-4> (0m18.17s).
-----> Setting up <mirror-maker-scripts-oracle-6-4>...
       Finished setting up <mirror-maker-scripts-oracle-6-4> (0m0.00s).
-----> Verifying <mirror-maker-scripts-oracle-6-4>...
       Preparing files for transfer
-----> Installing Busser (busser)
Fetching: thor-0.19.0.gem (100%)
Fetching: busser-0.7.1.gem (100%)
       Successfully installed thor-0.19.0
       Successfully installed busser-0.7.1
       2 gems installed
       Installing Busser plugins: busser-serverspec
       /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/installer.rb:388:in `ensure_required_ruby_version_met': rake requires Ruby version >= 2.0.0. (Gem::InstallError)
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/installer.rb:156:in `install'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:297:in `block in install'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:270:in `each'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:270:in `each_with_index'
       	from /opt/chef/embedded/lib/ruby/site_ruby/1.9.1/rubygems/dependency_installer.rb:270:in `install'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/rubygems.rb:44:in `install_gem'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:66:in `install_plugin_gem'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:53:in `install'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:43:in `block (2 levels) in install_all'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:43:in `each'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:43:in `block in install_all'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/rubygems.rb:71:in `silence_gem_ui'
       	from /tmp/verifier/gems/gems/busser-0.7.1/lib/busser/command/plugin_install.rb:42:in `install_all'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/command.rb:27:in `run'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:126:in `invoke_command'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `block in invoke_all'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `each'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `map'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:133:in `invoke_all'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/group.rb:232:in `dispatch'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:115:in `invoke'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:40:in `block in register'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/command.rb:27:in `run'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:126:in `invoke_command'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:359:in `dispatch'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:115:in `invoke'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:235:in `block in subcommand'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/command.rb:27:in `run'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/invocation.rb:126:in `invoke_command'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor.rb:359:in `dispatch'
       	from /tmp/verifier/gems/gems/thor-0.19.0/lib/thor/base.rb:440:in `start'
       	from /tmp/verifier/gems/gems/busser-0.7.1/bin/busser:8:in `<top (required)>'
       	from /tmp/verifier/bin/busser:23:in `load'
       	from /tmp/verifier/bin/busser:23:in `<main>'
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: 1 actions failed.
>>>>>>     Verify failed on instance <mirror-maker-scripts-oracle-6-4>.  Please see .kitchen/logs/mirror-maker-scripts-oracle-6-4.log for more details
>>>>>> ----------------------
```

Are you able to help me figure this out? I thought I'd see busser 0.7.0 installed, but actually saw busser 0.7.1.

I made this my test case after I tried installing rake 10.3.2 in the same Gemfile and seeing the same error.

Is this just because I'm using such an old version of chefdk, do you think?
